### ISSUE TITLE
Empty configuration in Prometheus after change to ConfigMap

### ISSUE THREAD

**Issue by wndhydrnt** [prometheus-operator/prometheus-operator#3677]
**What happened?**

After an update to the ConfigMap that contains alert and recording rules, the config of a Prometheus server at `/config` is empty:

```
global:
  scrape_interval: 1m
  scrape_timeout: 10s
  evaluation_interval: 1m
```

Triggering a reload of the Prometheus server via the `/-/reload` endpoint resolves the issue immediately.

We are running a lot of smaller Prometheus servers. This issue does not happen on every update of a ConfigMap but, given our number of Prometheus server, we run into it regularly.

**Did you expect to see something different?**

Correct config is loaded by a Prometheus server.

**How to reproduce it (as minimally and precisely as possible)**:

1. Update a ConfigMap (probably multiple times in a row).
2. Check `/config` endpoint of the Prometheus.

**Environment**

* Prometheus Operator version:

    `v0.35.1`
    

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:52:24Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:

  kops

* Manifests:

/

* Prometheus Operator Logs:

The only messages Prometheus Operator logs look like this:
```
level=info ts=2020-11-12T16:35:04.7058572Z caller=operator.go:1084 component=prometheusoperator msg="sync prometheus" key="<one of our prometheus servers>"
```

Logs of one of the Prometheus Pods where the reload failed:

```
Nov 12, 2020 @ 17:35:54.132 [prometheus] level=info ts=2020-11-12T16:35:54.132Z caller=main.go:799 msg="Loading configuration file" filename=/etc/prometheus/config_out/prometheus.env.yaml
Nov 12, 2020 @ 17:35:54.132 [rules-configmap-reloader] 2020/11/12 16:35:54 config map updated
Nov 12, 2020 @ 17:35:54.134 [rules-configmap-reloader] 2020/11/12 16:35:54 successfully triggered reload
Nov 12, 2020 @ 17:35:54.134 [prometheus] level=info ts=2020-11-12T16:35:54.134Z caller=main.go:827 msg="Completed loading of configuration file" filename=/etc/prometheus/config_out/prometheus.env.yaml
```

**Anything else we need to know?**:

#2501 sounds very similar to what we are observing.

**Comment by simonpasquier** [comment]
v0.35.1 is quite old. Please try with the latest prometheus operator version and report back if it's still happening.

**Comment by simonpasquier** [comment]
Sorry I didn't read carefully the first time but it is very likely that the issue has been fixed in v0.42 by #3457. Closing for now, feel free to reopen if you can reproduce with the latest version.
